db/db_impl_compaction_flush.cc

```diff
@@ -1169,6 +1240,38 @@ Status DBImpl::WaitForFlushMemTable(ColumnFamilyData* cfd,
   return s;
 }
 
+Status DBImpl::WaitForFlushMemTables(const autovector<ColumnFamilyData*>& cfds,
+    const autovector<uint64_t*>& flush_memtable_ids) {
+  Status s;
+  InstrumentedMutexLock l(&mutex_);
+  while (bg_error_.ok()) {
+    if (shutting_down_.load(std::memory_order_acquire)) {
+      return Status::ShutdownInProgress();
+    }
+    bool exit = true;
+    for (int i = 0; i < (int) cfds.size(); i++) {
+      if (cfds[i]->IsDropped()) {
+        s = Status::InvalidArgument("Cannot flush a dropped CF");
+        break;
+      }
+      if (cfds[i]->imm()->NumNotFlushed() > 0 &&
+          (flush_memtable_ids[i] == nullptr ||
```

Comment:
Can `flush_memtable_ids` store uint64_t rather than pointers to them? In the current code I didn't see how this can be `nullptr`.

Reply:
Now I tend to agree with you. With `atomic_flush=false`, flush is a single-CF operation. A null `flush_memtable_id` indicates that the current thread is willing to wait until all memtables of the column family are flushed (or error). With `atomic_flush=true`, it does not make sense for the current thread to wait for all memtables of a single column family to be flushed since different memtables of the same column family can belong to different atomic flushes.
Will update the code.